src/server/cloak-server.spec.js

```diff
@@ -91,10 +91,134 @@ describe('cloak server', function() {
+                    letters: [],
+                    consonantNum: 0,
+                    vowelNum: 0,
+                    disableConsonant: false,
+                    disableVowel: false
+                }
+            };
+
+            cloakConfig.room.newMember.bind(room, user)();
+
+            expect(user.message).toHaveBeenCalledWith('startGame');
+        });
+
+        it('sends resetLetters if the game was started in the joined room', () => {
+            users = [{name: 'Raul',data:{}},
+                     {name: 'Jamie',data:{}}];
+            rooms = [{id:'1'}];
+            cloak.getRooms.and.returnValue(rooms);
+            cloak.getRoom.and.returnValue({data:''});
+            room.getMembers.and.returnValue(users);
+            room.data = {
+                started: true,
+                leaderIndex: 0,
+                leaderId: '',
+                answering: false,
+                creator: {
+                    id: ''
+                },
+                scores: [],
+                letterList: {
+                    letters: ['a','b','c'],
+                    consonantNum: 0,
+                    vowelNum: 0,
+                    disableConsonant: false,
+                    disableVowel: false
+                }
+            };
+
+            cloakConfig.room.newMember.bind(room, user)();
+
+            expect(user.message).toHaveBeenCalledWith('resetLetters', ['a','b','c']);
+        });
+
+        it('sends startAnswering if the answering phase was started in the joined room', () => {
+            users = [{name: 'Raul',data:{}},
+                     {name: 'Jamie',data:{}}];
+            rooms = [{id:'1'}];
+            cloak.getRooms.and.returnValue(rooms);
+            cloak.getRoom.and.returnValue({data:''});
+            room.getMembers.and.returnValue(users);
+            room.data = {
+                started: true,
+                leaderIndex: 0,
+                leaderId: '',
+                answering: true,
+                answerTime: 30,
+                scores: [],
+                creator: {
+                    id: ''
+                },
+                letterList: {
+                    letters: ['a','b','c'],
+                    consonantNum: 0,
+                    vowelNum: 0,
+                    disableConsonant: false,
+                    disableVowel: false
+                }
+            };
+
+            cloakConfig.room.newMember.bind(room, user)();
+
+            expect(user.message).toHaveBeenCalledWith('startAnswering', 30);
+        });
+
+        it('sends stopAnswering if the game is not in the answering phase in the joined room', () => {
+            users = [{name: 'Raul',data:{}},
+                     {name: 'Jamie',data:{}}];
+            rooms = [{id:'1'}];
+            cloak.getRooms.and.returnValue(rooms);
+            cloak.getRoom.and.returnValue({data:''});
+            room.getMembers.and.returnValue(users);
+            room.data = {
```

Comment:
A lot of duplicate code in creating the room data. Would be cleaner to move this common code into a function.